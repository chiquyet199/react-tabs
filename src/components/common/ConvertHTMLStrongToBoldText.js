import React from "react";
import {Text} from "react-native";
import styled from "styled-components/native";

export const StyledView = styled(Text)`
  display: inline;
  flex-direction: row;
  flex-wrap: wrap;
  align-items: center;
  justify-content: center;
`;

export default ({htmlString}) => {
  const parsed = [];
  let cursor = 0;
  let searchingFor = "<strong>";
  while (cursor < htmlString.length) {
    let endPoint = htmlString.indexOf(searchingFor, cursor);
    endPoint = endPoint === -1 ? htmlString.length : endPoint;
    const token = htmlString.substring(cursor, endPoint);
    if (searchingFor === "<strong>") {
      parsed.push(<Text key={token}>{token}</Text>);
    } else {
      // Styled-components seems to ignore font-weight here
      parsed.push(
        <Text style={{fontWeight: "bold"}} key={token}>
          {token}
        </Text>,
      );
    }
    cursor = endPoint + searchingFor.length;
    searchingFor = searchingFor === "<strong>" ? "</strong>" : "<strong>";
  }

  return <StyledView>{parsed}</StyledView>;
};
