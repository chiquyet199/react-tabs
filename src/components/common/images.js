import styled from "styled-components/native";

export const CenterIconImage = styled.Image`
  width: 80px;
  height: 80px;
  margin-top: 20px;
  margin-bottom: 20px;
  align-self: center;
`;

export const CenterPageImage = styled.Image`
  width: 370px;
  height: 320px;
  margin-top: 20px;
`;
