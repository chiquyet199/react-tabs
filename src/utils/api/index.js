import * as example from "./exampleModule";
import * as users from "./userAuth";
import * as postcode from "./postcode";
import * as profile from "./profile";
import * as references from "./references";
import * as rentPassport from "./rentPassport";
import * as invites from "./invites";
import * as lease from "./lease";
import * as properties from "./properties";
import * as branches from "./branches";
import * as accounts from "./accounts";
import * as sharePassports from "./sharePassports";
import * as agency from "./agency";
import * as rentPassportAgencies from "./rentPassportAgencies";
import * as agents from "./agents";
import * as agencies from "./agencies";
import * as userNotifications from "./userNotifications";
import * as openBanking from "./openBanking";

export default {
  ...example,
  ...accounts,
  ...agencies,
  ...users,
  ...agents,
  ...postcode,
  ...profile,
  ...references,
  ...rentPassport,
  ...properties,
  ...sharePassports,
  ...invites,
  ...branches,
  ...agency,
  ...rentPassportAgencies,
  ...userNotifications,
  ...lease,
  ...openBanking,
};
