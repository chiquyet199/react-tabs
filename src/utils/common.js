import {Dimensions, Platform} from "react-native";
import React from "react";
import {connect} from "react-redux";
import styled from "styled-components/native";
import styledWeb, {withTheme} from "styled-components";

export const isWeb = Platform.OS === "web";
export const isNative = Platform.OS === "ios" || Platform.OS === "android";
export const isIos = Platform.OS === "ios";
export const isDev = process.env.NODE_ENV !== "production";
export const extractValueFromEvent = event =>
  event.target ? event.target.value : event;
export const extractKeyFromEvent = event =>
  event.nativeEvent ? event.nativeEvent.key : event.key;

export const sleep = ms => new Promise(resolve => setTimeout(resolve, ms));

export const cleanProps = props => {
  // List of attributes to skip.
  const BLACKLISTED_PROPS = ["dispatch"];
  // Initialize the `cleanProps` as an empty object.
  const ownCleanProps = {};
  // Iterate over all the properties (Eslint - loops should be avoided in favor of array iterations)
  Object.keys(props).forEach(propKey => {
    // Check if the key is not in the blacklisted array.
    if (BLACKLISTED_PROPS.indexOf(propKey) === -1) {
      // Append the attribute to the clean object.
      ownCleanProps[propKey] = props[propKey];
    }
  });

  // Return the cleaned properties.
  return ownCleanProps;
};

export const responsive = Component => styles => {
  const ResponsiveComponent = styled(Component)`
    ${({size}) => styles[size]};
  `;

  const ResponsiveWrapper = props => {
    const ownProps = cleanProps(props);

    return <ResponsiveComponent {...ownProps} />;
  };

  return connect(state => ({
    size: state.dimensions.size,
  }))(ResponsiveWrapper);
};

export const responsiveWeb = Component => styles => {
  const ResponsiveWebComponent = styledWeb(Component)`
    ${({size}) => styles[size]};
  `;

  const ResponsiveWebWrapper = props => {
    const ownProps = cleanProps(props);

    return <ResponsiveWebComponent {...ownProps} />;
  };

  return connect(state => ({
    size: state.dimensions.size,
  }))(ResponsiveWebWrapper);
};

export const isX = (() => {
  const d = Dimensions.get("window");
  const {height, width} = d;

  return (
    // This has to be iOS duh
    Platform.OS === "ios" &&
    // Accounting for the height in either orientation
    (height === 812 || width === 812)
  );
})();

// Use theme fallback because Selva has another reducer
export const themed = Component =>
  connect(({globaltheme, theme}) => ({
    globaltheme: globaltheme || theme,
  }))(withTheme(Component));

export const redirectToBaseRoute = (
  goTo,
  form,
  location = {},
  prefixOnly = false,
) => {
  const {pathname} = location;
  const urlArr = pathname.split("/").filter(i => i !== "");
  const [baseUrl] = urlArr;
  if (prefixOnly) {
    return baseUrl;
  }
  const {DynamicForm} = form;
  if (!DynamicForm) {
    goTo(`/${baseUrl}`);
  }

  return true;
};

export const parseStyle = style => {
  return Object.keys(style).reduce((previousValue, currentValue) => {
    const starter = previousValue ? `${previousValue};\n` : "";

    return `${starter}${currentValue}: ${style[currentValue]}`;
  }, "");
};

export const applyStyleObjectToStyledComponent = (style, component) => {
  if (!style || Object.keys(style).length === 0) {
    return component;
  }

  const parsedStyles = parseStyle(style);

  return styled(component)`
    ${parsedStyles};
  `;
};
