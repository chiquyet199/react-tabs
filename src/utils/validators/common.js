export const isRequired = (value, key) => !value && {[key]: "required"};

export const validate = async (fields, values) => {
  const formValidations = mapValidators(fields, values);
  const results = await Promise.all(formValidations);

  return createErrorObject(results);
};

const mapValidators = (fields, values) =>
  Object.keys(fields).reduce((acc, key) => {
    const {validations = []} = fields[key];

    return [
      ...acc,
      ...validations.map(runValidation(values[key], key, values)),
    ];
  }, []);

const runValidation = (value, key, values) => async validator =>
  (await validator(value, key, values)) || {};

const createErrorObject = results =>
  results.reduce(
    (acc, err) => ({
      ...acc,
      ...err,
    }),
    {},
  ); // returns empty object if no errors

export const atLeastOneSelectedIsRequired = (value, key, values) => {
  const selectedValues = Object.values(values);
  const arrayIsEmpty = selectedValues.length === 0;
  const valueIsSelected = selectedValues.some(
    selectedValue => selectedValue === true,
  );

  if (arrayIsEmpty || !valueIsSelected) {
    return {
      [key]: "selectAtLeastOne",
    };
  }

  return true;
};
