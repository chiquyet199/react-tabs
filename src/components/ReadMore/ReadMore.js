import React from "react";
import {string, object, number} from "prop-types";
import Truncate from "react-native-read-more-text";
import styled from "styled-components/native";
import {themed} from "../../utils/common";

export const Text = styled.Text`
  font-size: 16px;
  line-height: 24px;
  color: ${({theme, globaltheme}) => theme && theme[globaltheme].textColor};
  font-weight: ${({bold}) => (bold === "true" ? "bold" : "normal")};
  text-align: left;
  padding: 0;
`;

export const Link = styled(Text)`
  height: 24px;
  color: ${({theme, globaltheme}) => theme && theme[globaltheme].accentColor};
  font-weight: bold;
`;

export const Wrapper = styled(Text)`
  height: 24px;
`;

export const TruncatedWrapper = styled(Wrapper)`
  position: absolute;
  right: 0;
  bottom: 0;
  background-color: ${({theme}) => theme && theme.colors.white};
`;

export const ReadMore = ({text, lines, theme, globaltheme, more, less}) => {
  const themeProps = {theme, globaltheme};

  const renderTruncatedFooter = handlePress => {
    return (
      <TruncatedWrapper onPress={handlePress} {...themeProps}>
        ...
        <Link {...themeProps}>{more}</Link>
      </TruncatedWrapper>
    );
  };

  const renderRevealedFooter = handlePress => {
    return (
      <Wrapper onPress={handlePress} {...themeProps}>
        <Link {...themeProps}>{less}</Link>
      </Wrapper>
    );
  };

  return (
    <Truncate
      numberOfLines={lines}
      renderTruncatedFooter={renderTruncatedFooter}
      renderRevealedFooter={renderRevealedFooter}
    >
      <Text {...themeProps}>{text}</Text>
    </Truncate>
  );
};

ReadMore.defaultProps = {
  lines: 6,
  more: "read more",
  less: "show less",
};

ReadMore.propTypes = {
  text: string.isRequired,
  lines: number,
  less: string,
  more: string,
  globaltheme: string.isRequired,
  theme: object.isRequired,
};

export default themed(ReadMore);
