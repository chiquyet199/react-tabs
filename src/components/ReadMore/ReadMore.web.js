import React, {Component, Fragment} from "react";
import {string, object, number} from "prop-types";
import Truncate from "react-truncate";
import styled from "styled-components/native";
import {themed} from "../../utils/common";

export const Text = styled.Text`
  font-size: 16px;
  line-height: 24px;
  color: ${({theme, globaltheme}) => theme[globaltheme].baseColor};
  font-weight: ${({bold}) => (bold === "true" ? "bold" : "normal")};
  text-align: left;
  padding: 0;
`;

export const Link = styled(Text)`
  color: ${({theme, globaltheme}) => theme[globaltheme].accentColor};
  font-weight: bold;
  cursor: pointer;
`;

export class ReadMoreWeb extends Component {
  state = {
    expanded: false,
    truncated: false,
  };

  handleTruncate = truncated => {
    if (this.state.truncated !== truncated) {
      this.setState({
        truncated,
      });
    }
  };

  toggleLines = () => {
    this.setState({
      expanded: !this.state.expanded,
    });
  };

  render() {
    const {text, more, less, lines, theme, globaltheme} = this.props;
    const themeProps = {theme, globaltheme};
    const {expanded, truncated} = this.state;

    return (
      <Text {...themeProps}>
        <Truncate
          lines={!expanded && lines}
          ellipsis={
            <Text {...themeProps}>
              {"..."}
              <Link onClick={this.toggleLines} {...themeProps}>
                {more}
              </Link>
            </Text>
          }
          onTruncate={this.handleTruncate}
        >
          <Text {...themeProps}>{text}</Text>
        </Truncate>
        {!truncated &&
          expanded && (
            <Fragment>
              {" "}
              <Link onClick={this.toggleLines} {...themeProps}>
                {less}
              </Link>
            </Fragment>
          )}
      </Text>
    );
  }
}

ReadMoreWeb.defaultProps = {
  lines: 6,
  more: "read more",
  less: "show less",
};

ReadMoreWeb.propTypes = {
  text: string.isRequired,
  lines: number,
  less: string,
  more: string,
  globaltheme: string.isRequired,
  theme: object.isRequired,
};

export default themed(ReadMoreWeb);
