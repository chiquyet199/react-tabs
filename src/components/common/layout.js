import React from "react";
import styled from "styled-components/native";
import {isNative, isWeb, responsive, themed} from "../../utils/common";
import {TouchableOpacity} from "react-native";
import {DESKTOP_CONTAINER_WIDTH} from "../../constants/sizes";
import {ButtonContainer} from "../Button/Button";
import {func, node, number, string} from "prop-types";

export const FlexContainer = styled.View`
  flex: 1;
  width: 100%;
  height: 100%;
  max-width: ${({full}) => (full ? "100%" : `${DESKTOP_CONTAINER_WIDTH}px`)};
`;

export const Flexed = styled.View``;

export const Flex = styled.View`
  margin-top: ${({theme}) => theme.padding.top + "px"};
  position: absolute;
  width: 100%;
  background-color: transparent;
`;

export const FlexColumn = styled.View`
  max-width: 537px;
  display: flex;
  flex-direction: column;
  justify-content: center;
`;

export const TopBarContainer = styled.View`
  width: 100%;
  max-width: 100%;
  padding: 0;
  padding-top: ${({theme}) => theme.padding.top + "px"}
  z-index: 2;
  position: relative;
  background-color: ${({theme, globaltheme}) =>
    theme[globaltheme].topbar.background}
`;

export const BaseContainer = styled.View`
  flex: 1;
  padding: 0;
  width: 100%;
  z-index: 1;
`;

export const CenteredContainer = styled.View`
  flex: 1;
  justify-content: ${({align}) => align || "center"};
  align-self: center;
  align-items: center;
  width: 100%;
  max-width: ${({full}) => (full ? "100%" : `${DESKTOP_CONTAINER_WIDTH}px`)};
  margin-horizontal: auto;
  padding-horizontal: ${({full}) => (full === "true" ? "0" : "10px")};
  flex-basis: auto;
`;

export const EndPageContainer = responsive(CenteredContainer)({
  S: `justify-content: flex-start`,
});

export const PageContainer = responsive(styled(CenteredContainer)`
  width: 100%;
  max-width: ${({full}) =>
    full === "true" ? "100%" : `${DESKTOP_CONTAINER_WIDTH}px`};
  align-self: center;
  align-items: stretch;
  margin-top: ${({layout}) => (layout === "hq" ? "60px" : "40px")};
  ${({margintop}) => margintop >= 0 && `margin-top: ${margintop}px`};
  padding-horizontal: 10px;
  justify-content: center;
  flex-basis: auto;
`)({
  S: `
      padding-top: ${({theme}) => theme.padding.top + "px"};
      justify-content: space-between;
      padding-horizontal: 10px;
      margin-top: ${({layout}) => (layout === "hq" ? "80px" : "20px")};
    `,
});

export const FormContainer = responsive(styled.View`
  justify-content: center;
  align-items: center;
  max-width: ${DESKTOP_CONTAINER_WIDTH}px;
  width: 100%;
  align-self: center;
`)({
  S: `
      padding-top: 10px;
      justify-content: space-between;
      max-width: 100%;
    `,
});

export const FlexRow = styled.View`
  flex: 1;
  flex-direction: row;
  flex-basis: 100%;
`;

export const ButtonBar = styled.View`
  max-width: ${({isMenu}) =>
    isMenu ? "100%" : `${DESKTOP_CONTAINER_WIDTH}px`};
  width: 100%;
  border-top-color: ${({isMenu, theme}) =>
    isMenu ? theme.colors.lightSteel : theme.colors.none};
  border-top-width: ${({isMenu}) => (isMenu ? "1px" : 0)};
  background-color: ${({isMenu, theme}) =>
    isMenu ? theme.colors.white : theme.colors.none};
  flex-direction: row;
  margin-horizontal: 0;
  margin-bottom: ${({isMenu}) => (isMenu ? 0 : "5px")};
  flex-wrap: ${({wrap}) => (wrap ? "wrap" : "nowrap")};
  ${({mt}) => mt && `margin-top: ${mt}px}`};
  ${ButtonContainer} {
    margin: 20px;
  }
`;

export const BottomButtonBar = responsive(ButtonBar)({
  S: `flexGrow: 10;
      flexDirection: row;
      alignItems: center;
      justifyContent: flex-end;
    `,
});

export const ResponsiveWebScrollView = responsive(styled.ScrollView``)({
  M: `
        width: 500px;
  `,
  L: `
        width: 600px;
    `,
});

export const PageTitle = themed(styled.Text`
  width: 100%;
  font-size: 28px;
  line-height: 36px;
  color: ${({globaltheme, theme}) => theme[globaltheme].baseColor};
  text-align: ${({left}) => (left ? "left" : "center")};
  padding-horizontal: 5px;
  align-self: ${({left}) => (left ? "flex-start" : "center")};
  font-weight: ${({bold}) => (bold ? "bold" : "normal")};
  ${({mb}) => mb && `margin-bottom: ${mb}px}`};
  ${({color}) => color && `color: ${color}`};
`);

export const TextParagraph = themed(styled.Text`
  width: 100%;
  font-size: 16px;
  line-height: 24px;
  color: ${({disabled, theme, globaltheme}) =>
    disabled
      ? theme[globaltheme].disabledTextColor
      : theme[globaltheme].baseColor};
  font-weight: ${({bold}) => (bold ? "bold" : "normal")};
  font-style: ${({fontStyle}) => fontStyle || "normal"};
  align-self: ${({left}) => (left ? "flex-start" : "center")};
  text-align: ${({left}) => (left ? "left" : "center")};
  padding: ${({gutters}) => gutters || "15px 5px"};
  flex-shrink: 0;
  flex-grow: 0;
  ${({mb}) => mb && `margin-bottom: ${mb}px`};
  ${({color}) => color && `color: ${color}`};
`);

export const ViewParagraph = themed(styled.View`
  width: 100%;
  flex-shrink: 0;
  flex-grow: 0;
`);

export const SubTitle = themed(styled(TextParagraph)`
  font-weight: bold;
  color: ${({globaltheme, theme}) => theme[globaltheme].baseColor};
  ${({color}) => color && `color: ${color}`};
`);

export const TextLinkText = themed(styled.Text`
  text-align: center;
  font-size: 16;
  line-height: 24;
  color: ${({theme, globaltheme}) => theme[globaltheme].accentColor};
  font-weight: 700;
  padding-top: ${({gutters}) => (gutters >= 0 ? `${gutters}px` : "15px")};
  padding-bottom: ${({gutters}) => (gutters >= 0 ? `${gutters}px` : "15px")}
  padding-left: ${({gutters}) => (gutters >= 0 ? `${gutters}px` : "15px")}
  padding-right: ${({gutters}) => (gutters >= 0 ? `${gutters}px` : "15px")}
  text-align:  ${({align}) => align || "center"};
`);

export const TextLink = ({onClick, children, gutters, align, globaltheme}) => (
  <TouchableOpacity onPress={onClick}>
    <TextLinkText align={align} gutters={gutters} globaltheme={globaltheme}>
      {children}
    </TextLinkText>
  </TouchableOpacity>
);

TextLink.propTypes = {
  onClick: func,
  children: node,
  gutters: number,
  align: string,
  globaltheme: string,
};

export const NotchSafety = styled.View`
  height: 40px;
`;

export const PageHeader = styled.Text`
  font-size: 18;
  text-align: center;
  width: 100%;
  margin-top: ${({theme}) => theme.padding.top + 15 + "px"};
  color: ${({globaltheme, theme}) => theme[globaltheme].baseColor};
  position: absolute;
`;

export const HeadlineText = responsive(styled.Text`
  width: 100%;
  font-size: 28px;
  line-height: 36px;
  color: ${({globaltheme, theme}) => theme[globaltheme].baseColor};
  align-self: ${({left}) => (left ? "flex-start" : "center")};
  text-align: ${({left}) => (left ? "left" : "center")};
  margin: 8px 0 40px;
  padding: 0;
`)({
  S: `
    padding-left: 15px;
    `,
  L: `
    align-self: center;
   `,
});

export const BoxButtonBar = styled.View`
  align-self: center;
  align-items: stretch;
  width: 100%;
  margin-horizontal: auto;
  flex-direction: row;
`;

export const FixedWidthContainer = styled.View`
  width: ${DESKTOP_CONTAINER_WIDTH}px;
  margin-horizontal: auto;
  padding-horizontal: 15px;
`;

export const PaddedContent = styled.View`
  padding-horizontal: 10px;
`;

export const ErrorMessage = themed(styled(TextParagraph)`
  color: ${({theme}) => theme.colors.alertRed};
`);

// Scroll layout

export const ScrollContentContainer = styled.View`
  width: 100%;
  ${isNative ? "flex: 1;" : "height: calc(100vh - 52px)"};
  flex-grow: ${() => (isWeb ? 0 : 1)};
  flex-shrink: ${() => (isWeb ? 0 : 1)};
  justify-content: center;
  align-self: center;
  align-items: center;
`;

export const ScrollButtonBar = styled(ButtonBar)`
  padding-horizontal: 10px;
  ${({mt}) => mt && `margin-top: ${mt}px}`};
`;

export const ScrollScrollView = styled.ScrollView`
  ${isNative ? "" : "height: 100%"};
  width: 100%;
  flex-grow: 1;
  flex-shrink: 1;
`;
