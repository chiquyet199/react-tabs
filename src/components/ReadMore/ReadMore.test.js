import React from "react";
import {isWeb} from "../../utils/common";
import Truncate from "react-truncate";
import {shallow} from "enzyme";
import {extend} from "underscore";
import {ReadMore} from "./ReadMore";
import {ReadMoreWeb, Link} from "./ReadMore.web";

describe("Read More - page tests", () => {
  it("matches snapshot", () => {
    const expectedType = isWeb ? getWebComponent() : getComponent();
    expect(expectedType).toMatchSnapshot();
  });

  it("check if more/less is shown for Web and Native", () => {
    const {less, more} = getMockProps();
    const comp = isWeb ? getWebComponent() : getComponent();
    if (isWeb) {
      const ellipsis = comp.find(Truncate).prop("ellipsis");
      const lessLink = shallow(ellipsis).find(Link);
      expect(lessLink.prop("children")).toBe(more);
      lessLink.simulate("click");
      comp.update();
      expect(comp.find(Link).prop("children")).toBe(less);
    } else {
      const revealedFooter = shallow(comp.prop("renderRevealedFooter")());
      expect(revealedFooter.prop("children").toString()).toEqual(
        String(<Link>{less}</Link>),
      );
      const truncatedFooter = shallow(comp.prop("renderTruncatedFooter")());
      expect(truncatedFooter.prop("children")[1].toString()).toEqual(
        String(<Link>{more}</Link>),
      );
    }
  });

  const getComponent = props => {
    const parsedProps = extend(getMockProps(), props);

    return shallow(<ReadMore {...parsedProps} />);
  };

  const getWebComponent = props => {
    const parsedProps = extend(getMockProps(), props);

    return shallow(<ReadMoreWeb {...parsedProps} />);
  };

  const getMockProps = () => ({
    less: "LESS",
    more: "MORE",
    text:
      "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Assumenda beatae consequatur culpa cumque delectus distinctio est, fugiat illum laborum molestiae nisi obcaecati optio provident quasi, suscipit totam ullam ut voluptas?",
    lines: 3,
    theme: {
      light: {
        baseColor: "#6b7c93",
      },
      colors: {
        white: "#ffffff",
        canopySteel: "#6b7c93",
        canopyGreen: "#5ab88e",
      },
    },
    globaltheme: "light",
  });
});
