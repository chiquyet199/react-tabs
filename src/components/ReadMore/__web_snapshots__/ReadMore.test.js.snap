// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Read More - page tests matches snapshot 1`] = `
<Styled(Text)
  globaltheme="light"
  theme={
    Object {
      "colors": Object {
        "canopyGreen": "#5ab88e",
        "canopySteel": "#6b7c93",
        "white": "#ffffff",
      },
      "light": Object {
        "baseColor": "#6b7c93",
      },
    }
  }
>
  <Truncate
    ellipsis={
      <Styled(Text)
        globaltheme="light"
        theme={
          Object {
            "colors": Object {
              "canopyGreen": "#5ab88e",
              "canopySteel": "#6b7c93",
              "white": "#ffffff",
            },
            "light": Object {
              "baseColor": "#6b7c93",
            },
          }
        }
      >
        ...
        <Styled(Styled(Text))
          globaltheme="light"
          onClick={[Function]}
          theme={
            Object {
              "colors": Object {
                "canopyGreen": "#5ab88e",
                "canopySteel": "#6b7c93",
                "white": "#ffffff",
              },
              "light": Object {
                "baseColor": "#6b7c93",
              },
            }
          }
        >
          MORE
        </Styled(Styled(Text))>
      </Styled(Text)>
    }
    lines={3}
    onTruncate={[Function]}
    trimWhitespace={false}
    width={0}
  >
    <Styled(Text)
      globaltheme="light"
      theme={
        Object {
          "colors": Object {
            "canopyGreen": "#5ab88e",
            "canopySteel": "#6b7c93",
            "white": "#ffffff",
          },
          "light": Object {
            "baseColor": "#6b7c93",
          },
        }
      }
    >
      Lorem ipsum dolor sit amet, consectetur adipisicing elit. Assumenda beatae consequatur culpa cumque delectus distinctio est, fugiat illum laborum molestiae nisi obcaecati optio provident quasi, suscipit totam ullam ut voluptas?
    </Styled(Text)>
  </Truncate>
</Styled(Text)>
`;
